src/libcollections/slice.rs

```diff
@@ -1087,7 +1086,7 @@ impl<T> [T] {
     /// elements.
     ///
     /// This sort is stable and `O(n log n)` worst-case but allocates
-    /// approximately `2 * n`, where `n` is the length of `self`.
+    /// approximately `n / 2`, where `n` is the length of `self`.
```

Comment:
Saying "allocates `O(n / 2)`" is equivalent to "allocates less than `n / 2 * K`, for some constant `K`".

Saying "allocates approximately `n / 2`" is equivalent to "allocates between `n / 2 - K` and `n / 2 + K`", for some constant `K`".

The latter is more precise and therefore a better description of memory usage.